src/plugins/auto/executable_network.cpp

```diff
@@ -669,30 +670,80 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetMetric(const std::st
+                            } else {
+                                real = 4u;
+                            }
+                        } catch (const std::exception&) {
+                            LOG_WARNING("[AUTOPLUGIN] GetMetric RANGE_FOR_STREAMS failed");
+                            LOG_WARNING("[AUTOPLUGIN] use default value 4 for THROUGHPUT");
+                            real = 4u;
+                        }
+                        // if enable auto-batch, use below code
+                        // std::map<std::string, InferenceEngine::Parameter> options;
+                        // options["MODEL_PTR"] = _network.getFunction(); // CNNntework
+                        // try {
+                        //     auto optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
+                        //             METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
+                        //     auto rangeOfStreams = _core->GetMetric(deviceInfo.deviceName,
+                        //             METRIC_KEY(RANGE_FOR_STREAMS), options).as<std::tuple<unsigned int, unsigned int>>();
+                        //     real = (std::max)(real, std::get<1>(rangeOfStreams) * optimalBatchSize);
+                        // } catch (const InferenceEngine::Exception &iie) {
+                        //     LOG_WARNING("[AUTOPLUGIN]get optimal infer requset num for GPU auto-batch failed :%s", iie.what());
+                        // }
+                    } else {
+                        // if it is not THROUGHPUT mode, we think it's LATENCY mode
+                        real = 1u;
+                    }
+                } else if (deviceInfo.deviceName.find("VPUX") != std::string::npos) {
+                    if (bthroughput) {
+                        real = 8u;
+                    } else {
+                        real = 1u;
+                    }
+                } else if (deviceInfo.deviceName.find("MYRIAD") != std::string::npos) {
+                    if (bthroughput) {
+                        real = 4u;
+                    } else {
+                        real = 1u;
+                    }
+                } else if (deviceInfo.deviceName.find("CPU") != std::string::npos) {
+                    LOG_WARNING("[AUTOPLUGIN] not implement GetMetric for CPU_HELP,CPU case");
+                    LOG_WARNING("[AUTOPLUGIN] use default value 4 for THROUGHPUT, 1 for else");
+                    if (bthroughput) {
+                        real = 4u;
+                    } else {
+                        real = 1u;
+                    }
+                } else {
+                    LOG_WARNING("[AUTOPLUGIN] not implement GetMetric for device:%s,", deviceInfo.deviceName.c_str());
+                    LOG_WARNING("[AUTOPLUGIN] use default value 4 for THROUGHPUT, 1 for else");
+                    if (bthroughput) {
+                        real = 4u;
+                    } else {
+                        real = 1u;
```

Comment:
default real is 1u everywhere, shall we move it's value to be default (where the var is defined)

Reply:
Good point. We can move 1u as default of real which mean default 1 stream for latency mode for all HW.

Reply:
fix it